### DISCUSSION TITLE
Docker examples

### DISCUSSION THREAD

**Discussion by allen-woods** [framesurge/perseus#167] [Q&A]
I am currently working on refactoring the Dockerfiles in the deployment documentation and would like to update the examples used to better reflect the abilities of Perseus.

The changes I would like to make are as follows:

- 0.3.0-0.3.3
  - perseus-size-opt: examples/showcase
  - wee_alloc: examples/fetching
- 0.3.4-0.3.5
  - perseus-size-opt: core/state_generation
  - wee_alloc: demos/fetching
- 0.4.x
  - wee_alloc: core/state_generation
  - bonnie (branch): demos/fetching
- next
  - wee_alloc: core/state_generation
  - bonnie (branch): demos/fetching

Before I move forward with these changes, I'm seeking input regarding whether or not the alterations listed above are aligned with the community and the principles of the project.

Thanks!

**Comment by arctic-hen7** [comment]
Sorry, what exactly are you proposing here? Adding `wee_alloc` to some of the examples? Or adding some of the examples to the size optimizations plugin repo? Also, what do you mean by 'bonnie (branch)'?

**Reply by allen-woods** [reply]
Proposals:

- Migrate GitHub branch deployments (these implement `Bonnie`) to `0.4.x` and `next` versions only.
- Remove `perseus-size-opt` from `0.4.x` onward (deprecated).
- Provide manually configured `wee_alloc` implementation examples for all versions.
- Replace deployed examples in the current docs with `showcase` / `state_generation` and `fetching`.
  - This would mean no example deployment of `tiny`.

Let me know if you need further clarification 😊 Edit: or if any of this is anti-pattern 😅.

**Reply by arctic-hen7** [reply]
What do you mean by deployed examples and GitHub branch deployments?

Otherwise sounds great! (Sorry for the late reply, I've been absurdly busy the last week.)
